soql-analyzer/src/main/scala/com/socrata/soql/analyzer2/rewrite/RemoveSyntheticColumns.scala

```diff
@@ -0,0 +1,90 @@
+package com.socrata.soql.analyzer2.rewrite
+
+import com.socrata.soql.analyzer2
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.collection._
+
+class RemoveSyntheticColumns[MT <: MetaTypes] private (labelProvider: LabelProvider) extends StatementUniverse[MT] {
+  def rewriteStatement(stmt: Statement): Statement =
+    stmt match {
+      case ct@CombinedTables(_op, _left, _right) =>
+        if(ct.schema.values.exists(_.isSynthetic)) {
+          // there shouldn't ever actually be any synthetic columns to
+          // remove here, so this is "just in case".  We'll add a new
+          // wrapper select that drops the columns, rather than trying
+          // to rewrite left and right in an equivalent way.  Note we
+          // don't care about preserving our own labels, as this is
+          // the output schema for the analysis.
+
+          val newFrom = FromStatement(ct, labelProvider.tableLabel(), None, None)
+          Select(
+            Distinctiveness.Indistinct(),
+            OrderedMap() ++ ct.schema.iterator.flatMap { case (k, v) =>
+              if(v.isSynthetic) {
+                None
+              } else {
+                Some(labelProvider.columnLabel() -> NamedExpr(VirtualColumn[MT](newFrom.label, k, v.typ)(AtomicPositionInfo.Synthetic), v.name, v.hint, isSynthetic = false))
```

Comment:
Is the `AtomicPositionInfo.Synthetic` indicative of this column not having any *real* location in the underlying query since we're generating it here?

Reply:
Yep.  It's a different but related meaning of "synthetic" than the one in "synthetic column".  This means specifically "this expr does not correspond to any input soql source code"

Reply:
(vs. synthetic column which means "this column does not correspond to any explicitly-selected column in the soql source code", which is a subtle but distinct meaning since it's tied to column identity rather than source position)